torchaudio/datasets/librispeech.py

```diff
@@ -78,41 +96,20 @@ class LIBRISPEECH(Dataset):
     _ext_audio = ".flac"
 
     def __init__(
-        self, root: Union[str, Path], url: str = URL, folder_in_archive: str = FOLDER_IN_ARCHIVE, download: bool = False
+        self,
+        root: Union[str, Path],
+        url: str = URL,
+        folder_in_archive: str = FOLDER_IN_ARCHIVE,
+        download: bool = False,
     ) -> None:
+        if url not in _DATA_SUBSETS:
+            raise ValueError(f"Invalid url '{url}' given; please provide one of {_DATA_SUBSETS}.")
 
-        if url in [
-            "dev-clean",
-            "dev-other",
-            "test-clean",
-            "test-other",
-            "train-clean-100",
-            "train-clean-360",
-            "train-other-500",
-        ]:
-
-            ext_archive = ".tar.gz"
-            base_url = "http://www.openslr.org/resources/12/"
-
-            url = os.path.join(base_url, url + ext_archive)
-
-        # Get string representation of 'root' in case Path object is passed
         root = os.fspath(root)
+        self._path = os.path.join(root, folder_in_archive, url)
 
-        basename = os.path.basename(url)
-        archive = os.path.join(root, basename)
-
-        basename = basename.split(".")[0]
-        folder_in_archive = os.path.join(folder_in_archive, basename)
-
-        self._path = os.path.join(root, folder_in_archive)
-
-        if download:
-            if not os.path.isdir(self._path):
-                if not os.path.isfile(archive):
-                    checksum = _CHECKSUMS.get(url, None)
-                    download_url_to_file(url, archive, hash_prefix=checksum)
-                extract_archive(archive)
+        if download and not os.path.isdir(self._path):
+            download_librispeech(root, url)
```

Comment:
shall we add an exception if `download` is `False` and there is no `_path` in the local disk? something like https://github.com/pytorch/audio/blob/main/torchaudio/datasets/cmudict.py#L144